From: Yaroslav Halchenko <debian@onerussian.com>
Subject: ENH: make import of Selector also "optional" so nose doesn"t puke while buiding on 14.04

--- a/datalad/crawler/nodes/matches.py
+++ b/datalad/crawler/nodes/matches.py
@@ -15,15 +15,20 @@ import types
 
 from six import PY3
 
-from scrapy.selector import Selector
-from scrapy.http import Response
-
 from ...utils import updated
 from ...support.network import dlurljoin
 
 from logging import getLogger
 lgr = getLogger('datalad.crawler')
 
+from scrapy.http import Response
+try:
+    from scrapy.selector import Selector
+except ImportError:
+    lgr.debug("Failed to import Selector from scrapy, so matches would not be functional")
+    class Selector(object):
+        xpath = css = None
+
 
 # for now heavily based on scrapy but we might make the backend
 # replaceable
