From: Yaroslav Halchenko <debian@onerussian.com>
Subject: BF: adjusting version.py for 0.7.0 release + tune up to Changelog

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,7 +11,7 @@ We would recommend to consult log of the
 
 # 0.7.0 (Jun 25, 2017) -- when it works - it is quite awesome!
 
-I bet we will fix some bugs and make a world even a better place.
+New features, refactorings, and bug fixes.
 
 ### Major refactoring and deprecations
 
--- a/datalad/version.py
+++ b/datalad/version.py
@@ -13,7 +13,7 @@ import sys
 from os.path import lexists, dirname, join as opj, curdir
 
 # Hard coded version, to be done by release process
-__version__ = '0.6.0.dev1'
+__version__ = '0.7.0'
 
 # NOTE: might cause problems with "python setup.py develop" deployments
 #  so I have even changed buildbot to use  pip install -e .
